examples/usematches-loader-data/app/useMatchesData.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo } from "react";
+import { useMatches } from "remix";
+
+/**
+ * This base hook is used in other hooks to quickly search for specific data
+ * across all loader data using useMatches.
+ * @param {string} id The route id
+ * @param {string} key The object key (optional)
+ * @returns {unknown} The value of the key based on your currently loaded routes.
+ */
+export default function useMatchesData(id: string, key?: string): unknown {
```

Comment:
I don't personally like the `key` thing. It's not saving much from a typing perspective and it's adding API that people have to learn and understand.